rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java

```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
 	 */
 	@FixtureMethod
 	public boolean isViewVisible(String viewName) {
-		logger.trace("search for view with name: {}", viewName);
-		SWTBotView view = bot.viewByPartName(viewName);
+		logger.trace("search for view with titel: {}", viewName);
+		SWTBotView view = bot.viewByTitle(viewName);
 		return view.isActive();
 	}
 
+	/**
+	 * Selects an element in a tree. It takes a path string end tries to expand
+	 * the tree along the path. The node of the path is selectd.
+	 * 
+	 * @param locator
+	 *            of the tree.
+	 * @param itemName
+	 *            path to the item to be selectd. Example:
+	 *            "Root/SubNode/FinalNode"
+	 */
+	@FixtureMethod
+	public void selectElementInTreeView(String locator, String itemName) {
+		logger.trace("search for view with titel: {}", locator);
+		SWTBotTree tree = null;
+		if (locator.startsWith("[Single]")) {
+			tree = bot.tree();
+		} else {
+			SWTBotView view = bot.viewByTitle(locator);
+			tree = view.bot().tree();
+		}
+		assertNotNull(tree);
+		logger.trace("Open item with path: {}", itemName);
+		SWTBotTreeItem expandNode = tree.expandNode(itemName.split("/"));
+		assertNotNull(expandNode);
+		expandNode.select();
+	}
+
+	/**
+	 * Opens the context Menu of a widget and executes the menuitem described as
+	 * a path.
+	 * 
+	 * @param viewName
+	 *            id of the view / widget with the context menu
+	 * @param menuItem
+	 *            poath to the menuitem Example: "New/Project"
+	 * @throws Exception
+	 *             on failure to break the test.
+	 */
+	@FixtureMethod
+	public void executeContextMenuEntry(String viewName, String menuItem) throws Exception {
+		logger.trace("search for view with titel: {}", viewName);
+		SWTBotView view = bot.viewByTitle(viewName);
+		assertNotNull(view);
+		SWTBotTree tree = view.bot().tree();
+		assertNotNull(tree);
+		MenuItem item = ContextMenuHelper.contextMenu(tree, menuItem.split("/"));
+		new SWTBotMenu(item).click();
+		logger.trace("Yeaa renne weiter");
+	}
+
+	/**
+	 * Looking for a widget in that is typeable and is identifiable by the
+	 * locator.
+	 * 
+	 * @param locator
+	 *            of the widget.
+	 * @param value
+	 *            set to the widget.
+	 */
+	@FixtureMethod
+	public void typeInto(String locator, String value) {
+		logger.trace("search for text with titel: {}", locator);
+		SWTBotText text = null;
+		if (locator.startsWith("[Single]")) {
+			text = bot.text();
+		}
```

Comment:
What happens if the locator does not start with `[Single]`? Shouldn't the locator be passed to the swt bot?